### PR TITLE
Port OldIndexBackwardsCompatibilityIT#assertBasicSearchWorks over to full cluster restart qa module

### PR DESCRIPTION
**Pull request by martijnvg**
Relates to #24939

### CODE REVIEW

File: qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java [hunk 1]
```diff
@@ -64,6 +69,96 @@ protected boolean preserveReposUponCompletion() {
...
+            for (int doc = 0; doc < numDocs; doc++) {
+                XContentBuilder docBuilder = jsonBuilder();
+                docBuilder.startObject();
+                docBuilder.field("string", Integer.toString(randomInt(100)));
+                docBuilder.field("long_sort", (long) randomInt(100));
+                docBuilder.field("double_sort", (double) randomInt(100));
+                docBuilder.field("bool", randomBoolean());
+                docBuilder.field("field.with.dots", Integer.toString(randomInt(100)));
+                docBuilder.endObject();
+                client().performRequest("PUT", "/index/doc/" + Integer.toString(doc), Collections.emptyMap(),
```
Comment (nik9000) [inline qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java]: I don't think you need the `Integer.toString` bit.

File: qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java [hunk 2]
```diff
@@ -64,6 +69,96 @@ protected boolean preserveReposUponCompletion() {
...
+
+        searchRequestBody = "{ \"query\": { \"exists\" : {\"field\": \"field.with.dots\"} }}";
+        response = consume(client().performRequest("GET", "/index/_search", Collections.emptyMap(),
+            new StringEntity(searchRequestBody, ContentType.APPLICATION_JSON)));
+        assertNoFailures(response);
+        numDocs2 = (int) XContentMapValues.extractValue("hits.total", response);
+        assertEquals(numDocs1, numDocs2);
+    }
+
+    static Map<String, Object> consume(Response response) throws IOException {
```
Comment (nik9000) [inline qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java]: Maybe `Map<String, Object> toMap(Response response)`?

File: qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java [hunk 3]
```diff
@@ -64,6 +69,96 @@ protected boolean preserveReposUponCompletion() {
         return true;
     }
 
+    public void testOldIndexBwc() throws Exception {
```
Comment (nik9000) [inline qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java]: Maybe `testSearch` instead?

I think it'd be nice to move some of the random document generation from `testRandomDocumentsAndSnapshot` below into this and just use `testRandomDocumentsAndSnapshot` for testing the translog stuff.

File: qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java [hunk 4]
```diff
@@ -64,6 +69,96 @@ protected boolean preserveReposUponCompletion() {
...
+
+    void assertBasicSearchWorks() throws IOException {
+        logger.info("--> testing basic search");
+        Map<String, Object> response = consume(client().performRequest("GET", "/index/_search"));
+        assertNoFailures(response);
+        int numDocs1 = (int) XContentMapValues.extractValue("hits.total", response);
+        logger.info("Found {} in old index", numDocs1);
+
+        logger.info("--> testing basic search with sort");
+        String searchRequestBody = "{ \"sort\": [{ \"long_sort\" : \"asc\" }]}";
```
Comment (nik9000) [inline qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java]: Maybe it should just be `long` instead of `long_sort`.

**Review by nik9000 — APPROVED** [review]
I left a few suggestions for improvement or followups.

Can you remove the `OldIndexBackwardsCompatibilityIT#assertBasicSearchWorks` method so we're sure we've ported it?

**Comment by martijnvg** [comment]
@nik9000 I've updated the PR, based on your feedback.

File: qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java [hunk 5]
```diff
@@ -64,6 +70,95 @@ protected boolean preserveReposUponCompletion() {
...
+                mappingsAndSettings.endObject();
+                mappingsAndSettings.endObject();
+                mappingsAndSettings.endObject();
+            }
+            mappingsAndSettings.endObject();
+            client().performRequest("PUT", "/index", Collections.emptyMap(),
+                new StringEntity(mappingsAndSettings.string(), ContentType.APPLICATION_JSON));
+
+            int numDocs = randomIntBetween(2000, 3000);
+            indexRandomDocuments("index", numDocs, false, i -> {
```
Comment (nik9000) [inline qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java]: We can allow flush here, I think.

**Review by nik9000 — APPROVED** [review]
Left a minor but otherwise LGTM.

**Pull request merged**